crypto/rand/randfile.c

```diff
@@ -83,26 +83,37 @@ int RAND_load_file(const char *file, long bytes)
         return 0;
 
 #ifndef OPENSSL_NO_POSIX_IO
-    if (stat(file, &sb) < 0 || !S_ISREG(sb.st_mode)) {
-        RANDerr(RAND_F_RAND_LOAD_FILE, RAND_R_NOT_A_REGULAR_FILE);
-        ERR_add_error_data(2, "Filename=", file);
-        return -1;
-    }
+    /*
+     * If not a regular file, assume it's a device and don't try to
+     * read everything.
+     */
+    if (stat(file, &sb) >= 0 && !S_ISREG(sb.st_mode) && bytes < 0)
+        bytes = 256 / 8;
```

Comment:
I would not do that at all. -1 is defined in the manual as read everything. This does not work with non-regular files like /dev/urandom. There I suggest to reject it. Could have this work before?